helix-term/src/commands.rs

```diff
@@ -2934,27 +2934,40 @@ fn open(cx: &mut Context, open: Open) {
             Open::Above => cursor_line,
         };
 
+        let line_num = new_line.saturating_sub(1);
+        let text = doc.text().slice(..);
+
         // Index to insert newlines after, as well as the char width
         // to use to compensate for those inserted newlines.
         let (line_end_index, line_end_offset_width) = if new_line == 0 {
             (0, 0)
         } else {
             (
-                line_end_char_index(&doc.text().slice(..), new_line.saturating_sub(1)),
+                line_end_char_index(&text, line_num),
                 doc.line_ending.len_chars(),
             )
         };
 
+        let line_start_index = doc.text().line_to_char(line_num);
+
+        let indent_query_pos =
+            if !Range::new(line_start_index, line_end_index).is_single_grapheme(text) {
+                graphemes::prev_grapheme_boundary(text, line_end_index)
```

Comment:
If I understood you correctly you are purposefully starting the capture at the start of the newline char instead of the end here, right? In that case a comment would be nice.

Reply:
No, I'm starting it at the last character in the line, before the line feed. This way, it is more likely to start with a deeper node, such as an identifier. Currently, it's starting the indent query from the line feed, which is often a big scope. I will add a comment.